main.go

```diff
@@ -28,6 +30,33 @@ func init() {
 	httpcaddyfile.RegisterHandlerDirective("dynamic_proxy", parseCaddyfile)
 }
 
+type AuthSite struct {
+	To   string
+	Path string
+}
+
+func NewAuthSite(site string) (AuthSite, error) {
```

Comment:
My feeling is that a `New____` function (a/k/a Contstructor) shouldn't return an error, though I don't have a source to back me up on that. If you would like to try this, I recommend splitting this into two functions:
```go
func NewAuthSite(siteURL *url.URL) *AuthSite {
	result := AuthSite{
		To:   siteURL.Host,
		Path: siteURL.Path,
	}
	if siteURL.Port() == "" {
		result.To += ":80"
	}
	return result
}
```
and
```go
func toURL(addr string) (*url.URL, error) {
  	if m, _ := regexp.MatchString("^\\w+://", addr); !m {
		site = "http://" + site
	}
	return url.Parse(addr)
}
```

Reply:
No longer have a `New___` function now parsing directly from CLI